geode-core/src/main/java/org/apache/geode/internal/net/NioFilter.java

```diff
@@ -53,7 +53,7 @@ ByteBuffer ensureWrappedCapacity(int amount, ByteBuffer wrappedBuffer,
    * wrappedBuffer = filter.ensureWrappedCapacity(amount, wrappedBuffer, etc.);<br>
    * unwrappedBuffer = filter.readAtLeast(channel, amount, wrappedBuffer, etc.)
    */
-  ByteBuffer readAtLeast(SocketChannel channel, int amount, ByteBuffer wrappedBuffer)
+  ByteBuffer readAtLeast(int amount, ByteBuffer wrappedBuffer, Socket socket)
```

Comment:
that's explained in my initial comment

Reply:
on second thought, I'm going to undo those changes.  My first cut at a solution modified the filters to allow their use with an SSLSocket and its input stream, but I decided that it was better to stop using them at all.

Reply:
ah I failed to understand that both these things are true at once:

"This PR … removes the use of "new IO" SSLEngine as an option."

and

"There are a few changes to the NIO "engine" implementations that I made
while testing performance that I decided to keep because they allow the
use of NioPlainEngine with SSLSockets. That could be an advantage when
performing similar work in client/server communications"

If I understand: implication of the second bit is that we didn't remove SSLEngine from the codebase—we only eliminated it in the P2P messaging path.